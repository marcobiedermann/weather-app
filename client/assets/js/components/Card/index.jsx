import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import styles from './style.module.css';

const Card = props => {
  const { children, className, ...otherProps } = props;

  return (
    <div className={classNames(className, styles.card)} {...otherProps}>
      {children}
    </div>
  );
};

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

Card.defaultProps = {
  children: null,
  className: '',
};

export default Card;
