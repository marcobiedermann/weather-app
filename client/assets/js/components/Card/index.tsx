import React, { FC } from 'react';
import styles from './style.module.css';

const Card: FC = (props) => {
  return <div className={styles.card} {...props} />;
};

export default Card;
