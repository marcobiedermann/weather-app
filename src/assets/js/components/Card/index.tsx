import React, { FC } from 'react';
import styles from './style.module.css';

const Card: FC = (props) => <div className={styles.card} {...props} />;

export default Card;
