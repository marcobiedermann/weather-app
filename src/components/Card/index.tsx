import React, { ReactNode } from 'react';
import styles from './style.module.css';

interface CardProps {
  children: ReactNode;
}

function Card(props: CardProps): JSX.Element {
  return <div className={styles.card} {...props} />;
}

export type { CardProps };
export default Card;
