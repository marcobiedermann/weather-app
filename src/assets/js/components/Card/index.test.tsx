import React from 'react';
import renderer from 'react-test-renderer';
import Card from '.';

describe('components/Card', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const tree = renderer.create(<Card>Content</Card>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
