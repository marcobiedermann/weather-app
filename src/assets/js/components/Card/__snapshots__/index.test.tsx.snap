// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Card renders correctly 1`] = `
<div
  className="card"
>
  Content
</div>
`;
